codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/RestXmlProtocolGenerator.java

```diff
@@ -0,0 +1,96 @@
+package software.amazon.smithy.aws.go.codegen;
+
+import java.util.Set;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.go.codegen.GoStackStepMiddlewareGenerator;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.integration.HttpBindingProtocolGenerator;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.TimestampFormatTrait;
+
+abstract class RestXmlProtocolGenerator extends HttpBindingProtocolGenerator {
```

Comment:
This class will also want to implement the protocol tests

```java
    @Override
    public void generateProtocolTests(GenerationContext context) {
        new HttpProtocolTestGenerator(context,
                new HttpProtocolUnitTestRequestGenerator.Builder(),
                new HttpProtocolUnitTestResponseGenerator.Builder(),
                new HttpProtocolUnitTestResponseErrorGenerator.Builder()
        ).generateProtocolTests();
    }
```

Reply:
Added stub for HttpProtocolTestGenerator. This leads to Go vet error for protocol test deserializers where response is declared but not used. 

TODO:
- [ ] Fix those unused response var error

Reply:
Fixed by adding a status code check for response.